Added inverted behavior - piping JSON files into the same template

Tobsta opened the pull request:
I've implemented the feature I thought this plugin was lacking in #44.

Basically, now, as well as the previous behavior, there is now inverted behaviour. This means that you can do the following: (extracted from the updated README)

> ### Inverted behaviour
> This example takes each `.json` file in the current directory, and uses the same template for each. This is useful for, say, building a series of error pages with the same styles but just slightly different wording.
> ```js
> gulp.task('template-path', function() {
> 	return gulp.src('./*.json')
> 		.pipe(mustache('example.mustache', {extension: '.html'}))
> 		.pipe(gulp.dest('./'));
> });
> ```

I tried to be consistent with your code style, but I don't have much experience with open source so forgive me if I've made any silly mistakes.

rogeriopvl commented:
Thanks for the PR @Tobsta 

I'll give it a closer look as soon as possible. Meanwhile can you fix the build? it's breaking on travis. Also your code editor is not adding newlines at the end of file, this needs to be fixed to be accepted.

Tobsta commented:
Yep, I'll have a look at that error, and see if I can incorporate testing for the new feature into the mocha test file already there. Will probably update tomorrow, dunno if I'll have time to finish rn.

Tobsta commented:
@rogeriopvl There we go, fixed up that failed test and added three more tests. I've removed the test2 directory (which I was using that for manual testing) and the documentation has been updated.

rogeriopvl commented on .gitignore:
```diff
@@ -1,3 +1,3 @@
 .DS_Store
 node_modules/
-temp/
+temp/
```
This file is still missing a newline at the end

rogeriopvl commented:
@Tobsta after carefully considering this PR, I feel that this feature is out of the scope for this plugin, makes the code a lot more complex and maybe it belongs in a separate plugin, or can be achieved by using [gulp-data](https://www.npmjs.com/package/gulp-data).

Let me know if and why you disagree.
